scifin/classifier/classifier.py

```diff
@@ -259,44 +311,65 @@ def kmeans_advanced_clustering(corr, names_features=None, max_num_clusters=None,
     -----
       Function adapted from "Machine Learning for Asset Managers",
       Marcos López de Prado (2020).
+
+      To learn more about sklearn.cluster.KMeans():
+      https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html#sklearn.cluster.KMeans
     """
     
     # Checks
     if (max_num_clusters is not None) and (not isinstance(max_num_clusters, int)):
         raise AssertionError("max_num_clusters must be integer.")
-    if not isinstance(n_init, int):
-        raise AssertionError("n_init must be integer.")
       
     # Initializations
     if max_num_clusters==None:
         max_num_clusters = corr.shape[1]-1
+    if names_features is None:
+        names_features = corr.columns.tolist()
+
     # Using base clustering as initial step
     corr1, clusters, silh = kmeans_base_clustering(corr,
                                                    names_features=names_features,
                                                    max_num_clusters=min(max_num_clusters, corr.shape[1]-1),
-                                                   n_init=n_init)
+                                                   **kwargs)
     
     # Compute t-stat for each cluster
     cluster_tstats = {i: np.mean(silh[clusters[i]]) / np.std(silh[clusters[i]]) for i in clusters.keys()}
-    # Obtain the mean over clusters
+
+    # Obtain the mean t-stat over clusters
+    print(cluster_tstats)
     tstat_mean = sum(cluster_tstats.values()) / len(cluster_tstats)
-    # Select the clusters which have a t-stat below the mean
-    redo_clusters = [i for i in cluster_tstats.keys() if cluster_tstats[i] < tstat_mean]
 
-    # Only one cluster
-    if len(redo_clusters)<=1:
+    # Select the clusters having a t-stat below the mean
+    clusters_to_redo = [i for i in cluster_tstats.keys() if cluster_tstats[i] < tstat_mean]
+
+    # Only one cluster to redo, nothing to do
+    if len(clusters_to_redo)<=1:
         return corr1, clusters, silh
-    # More than one cluster
+
+    # More than one cluster to redo
     else:
```

Comment:
else is actually not needed here. But you may keep it if you think that makes the code more legible

Reply:
You're right. Let's keep it for structure (I'm still looking for a nice way to present comments in a hierarchical way... for example if I remove the "else" I would need to present the comments "# More than one cluster to redo" and "# Get the key name of concerned clusters" into some kind of hierarchical relashionship).